scripts/make-certs.sh

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env bash 
+
+set -e 
+
+if ! command -v openssl &> /dev/null
+then
+    echo "openssl is required to create certificates"
+    exit 
+fi
+
+mkdir ./certs 2> /dev/null
+cd ./certs 
+
+cat > dev-server.cnf << EOF
+[req]
+default_md = sha256
+prompt = no
+req_extensions = v3_ext
+distinguished_name = req_distinguished_name
+
+[req_distinguished_name]
+CN = localhost
+
+[v3_ext]
+keyUsage = critical,digitalSignature,keyEncipherment
+extendedKeyUsage = critical,serverAuth,clientAuth
+subjectAltName = DNS:localhost
+subjectAltName = IP:0.0.0.0
```

Comment:
What does 0.0.0.0 mean in this context? For servers it just specifies listen-on-all-interfaces, right?

Reply:
Yeah, it still means 'this host' and the general interpretation is list on all interfaces.  It's because I found I was binding/requesting both to 127.0.0.1 and 0.0.0.0 arbitrarily, and so we cover both cases.  In addition we use localhost in case that's defined in someone's host file.